File: backend/Api/Data/Configurations/Supplier.cs
using Microsoft.EntityFrameworkCore;
using Microsoft.EntityFrameworkCore.Metadata.Builders;
using Api.Models;

namespace Api.Data.Configurations
{
    public class SupplierConfiguration : IEntityTypeConfiguration<Supplier>
    {
        public void Configure(EntityTypeBuilder<Supplier> builder)
        {
            builder.ToTable("Suppliers");
            builder.HasKey(s => s.Id);
            builder.Property(s => s.Name).IsRequired().HasMaxLength(200);
            builder.Property(s => s.ContactPerson).HasMaxLength(100);
            builder.Property(s => s.Phone).IsRequired().HasMaxLength(20);
            builder.Property(s => s.Email).IsRequired().HasMaxLength(100);
            builder.HasIndex(s => s.Email).IsUnique();
            builder.Property(s => s.Address).HasMaxLength(500);
            builder.Property(s => s.Category).HasMaxLength(100);
            builder.Property(s => s.Rating).HasColumnType("decimal(3,2)");
            builder.Property(s => s.IsActive).IsRequired().HasDefaultValue(true);
            builder.Property(s => s.CreatedAt).IsRequired();

            builder.HasMany(s => s.PurchaseOrders)
                .WithOne(po => po.Supplier)
                .HasForeignKey(po => po.SupplierId)
                .OnDelete(DeleteBehavior.Restrict);
        }
    }
}

File: frontend/src/app/core/domain/supplier/supplier.ts
/**
 * Represents a Supplier
 */
export interface Supplier {
  /** ID */
  id: number;

  /** Name - supplier name */
  name: string;

  /** Contact Person - main contact */
  contactPerson?: string | null;

  /** Phone - contact phone */
  phone: string;

  /** Email - contact email */
  email: string;

  /** Address - supplier address */
  address?: string | null;

  /** Category - supplier category/type */
  category?: string | null;

  /** Rating - supplier rating (0-5) */
  rating?: number;

  /** Is Active - is the supplier active */
  isActive?: boolean;

  /** Created At */
  createdAt?: Date;

  /** Updated At */
  updatedAt?: Date;

  /** Deleted At */
  deletedAt?: Date | null;
}
